### DISCUSSION TITLE
Cannot convert value of type - Examples will not compile using Xcode 13.3

### DISCUSSION THREAD

**Discussion by iwt-sebastian-boldt** [pointfreeco/episode-code-samples#113] [Q&A]
I started working and wrapping my head around the whole TCA Topic and created a project by myself where I used 
Swift Packages instead of Frameworks and I also structured the code in a way that fits my needs. 

Sadly suddenly some of the approaches you guys explained in your video series does not seem to compile.
First I thought I was an issue introduced by myself but
it seems that the example projects also do not compile under Xcode 13.3 for some weird reasons.

https://github.com/pointfreeco/episode-code-samples/tree/main/0076-effectful-state-management-synchronous-effects

Do you guys have any Idea what's going on here?

<img width="909" alt="Screenshot 2022-04-29 at 11 26 58" src="https://user-images.githubusercontent.com/88190290/165918957-0053c938-a675-498b-b7db-37a935d0aa9d.png">

**Comment by mbrandonw** [comment]
I believe this code is not compiling simply because in the episode we did not end in a compiling state. The entire series of effectful statement management is basically one big refactor until we get to the final form of `Effect`, and so it's not until the last episode, [#79](https://github.com/pointfreeco/episode-code-samples/tree/main/0079-effectful-state-management-wtp), that things fully compile.

That's a little unfortunate, but there are small changes you can make to 76, 77 and 78 to get them compiling if you want to play around with the code at those stages. For example, in 76 you can just make sure all the reducers have a return type of `Effect` and then add a `return {}` at the end of the body of the reducer. That should get you most of the way to compiling, but there may be a few small other things to fix too.

**Reply by iwt-sebastian-boldt** [reply]
I am actually not sure if you are right because inside the corresponding Episode-Video everything seems to compile. 
The Solution you proposed also does not seem feasible to me because there is currently no Effect Type defined in the Project.

**Comment by mbrandonw** [comment]
Also I'm going to convert this issue to a discussion, but please let us know if you have more questions!
